### ISSUE TITLE
How can I correctly link to subheadings in markdown format?

### ISSUE THREAD

**Issue by MatthiasKauer** [vimwiki/vimwiki#1356]
Hi,
I am trying to make internal links or anchors within a single file work but some don't behave as expected.
Hopefully, the following example file demonstrates what I mean:
```
Does vimwiki have issues with level 2 links?
--------------------------------------------

# Problem
Top links like [this](#TopCategoryOne) or [that](#TopCategoryTwo) jump to the respective anchor. Great!

Links to sub categories seem to jump to the first anchor:
- [Here, we correctly jump to T1-S2](#TopCategoryOne#SubCategoryTwo)
- [Here, however, we incorrectly land at T1-S2](#TopCategoryTwo#SubCategoryTwo)
- [T2-S3 works, likely because T1-S3 doesn't exist](#TopCategoryTwo#SubCategoryThree)

# TopCategoryOne
## SubCategoryOne
## SubCategoryTwo


# TopCategoryTwo
## SubCategoryOne
## SubCategoryTwo
## SubCategoryThree
```
When I say "jump", I mean "press Enter". Not sure what command that corresponds to.

Regarding other questions from your form:
The VimWiki part from my `.vimrc` looks like this. I don't really recall why. None of my files are under `~/.wiki` :open_mouth:; except for this vimwiki is working well and I use it a lot. Thank you so much for creating it!
```
Plug 'vimwiki/vimwiki'  " \ww for personal wiki and markdown highlighting
let g:vimwiki_list=[{'path': '~/.wiki', 'syntax': 'markdown', 'ext': '.md', 'auto_tags': 1}]
let g:vimwiki_ext2syntax = {'.md': 'markdown', '.markdown': 'markdown', '.mkd': 'markdown'}
let g:vimwiki_conceallevel=0
let g:vimwiki_auto_chdir=1  " change to wiki dir when opening a page; maybe simplifies search
let g:vimwiki_folding='expr'    " fast, but only folds sections and blocks
```

Include the output of `:VimwikiShowVersion`.
```
Version: 2022.12.02                                                                                                                                                                                                                           
Os: Linux                                                                                                                                                                                                                                     
Vim: 801                                                                                                                                                                                                                                      
Branch: master                                                                                                                                                                                                                                
Revision: fea8bee                                                                                                                                                                                                                             
Date: 2022-12-04 21:25:52 -0700        
```

**Comment by tinmarino** [comment]
Hi @MatthiasKauer,

Thank you for the report. Indeed, there have been a bug in the markdown link jumper for at least 3 years. I did not see it.

Thank you for the example: it helped created regression tests. I hope this does not break some functionality in wiki syntax but anyway, with the regression tests, it is permanently fixed for markdown.

Cheers.

**Comment by tinmarino** [comment]
Closing <= Fixed by 09804c2a => please rebase to the new dev head.
